src/components/misc/ScrollToTop.jsx

```diff
@@ -0,0 +1,12 @@
+import { useEffect } from "react";
+import { useLocation } from "react-router-dom";
+
+export default function ScrollToTop() {
+  const { pathname } = useLocation();
+
+  useEffect(() => {
```

Comment:
You probably want to setup a listener hear instead something like this.
And then do `useHistory` to get the `history` https://stackoverflow.com/questions/36904185/react-router-scroll-to-top-on-every-transition

Reply:
I believe the implemented solution is the correct one for `react-router-dom` versions 6+ as the `useHistory` hook was removed. The ScrollToTop component I rolled with is taken from a later answer in the link you provided: https://stackoverflow.com/a/61602724